src/main/java/com/cityelf/repository/AddressesRepository.java

```diff
@@ -36,4 +36,15 @@
   List<Address> findSimilarAddress(String address, String number);
 
   List<Address> findByAddressInOrAddressUaIn(List<String> addresses, List<String> addressesUa);
+
+  @Query(value = "select distinct SUBSTRING_INDEX(tmp,' [',1) as unique_names "
```

Comment:
Let's discuss this at tomorrow's meeting. We'd better not use native queries if it is possible.